### PR TITLE
[FLINK-8984][network] Drop taskmanager.exactly-once.blocking.data.enabled config option

### PR DESCRIPTION
**Pull request by pnowojski**
Previously there were twe options:

taskmanager.network.credit-based-flow-control.enabled

and

taskmanager.exactly-once.blocking.data.enabled

If we disabled first one, but keept default value for the second one deadlocks will occur.

By dropping taskmanager.exactly-once.blocking.data.enabled we can always use:
 - blocking BarrierBuffer for credit based flow control
 - spilling BarrierBuffer for non credit based flow control.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by zhijiangW** [comment]
Thanks piotr, I agree with it.

**Comment by zentol** [comment]
What would happen if `taskmanager.network.credit-based-flow-control.enabled` is enabled, but `taskmanager.exactly-once.blocking.data.enabled` is disabled?

Is this an invalid setting?

**Comment by pnowojski** [comment]
Yes, it's a valid setting, but has worse performance and whole credit based flow control looses it's sense. `taskmanager.exactly-once.blocking.data.enabled` setting was thought as a fallback option if there is a bug in new code path, but this is achieved by using `taskmanager.network.credit-based-flow-control.enabled` in two places (and both of those places are about using `credit-based-flow-control`).

In other words, we do not think there is a much value in allowing `taskmanager.network.credit-based-flow-control.enabled` enabled, but `taskmanager.exactly-once.blocking.data.enabled` 
disabled, and it would make the configuration more complex/confusing for the users.

**Comment by zentol** [comment]
merging.

**Comment by pnowojski** [comment]
Thanks!
